src/functions.js

```diff
@@ -0,0 +1,110 @@
+// select variables
+const todoListElement = document.getElementById('todo');
+let todos = JSON.parse(localStorage.getItem('todos')) || [];
+const todoInput = document.querySelector('input');
+let EditTodoId = -1;
+
+// save Todo function to the local storage of a user machine
+
+export function saveTodo() {
+  const todoValue = todoInput.value;
+  const isEmpty = todoValue === '';
+  // check duplicate input
+  const isDuplicate = todos.some((todo) => todo.value.toUpperCase() === todoValue.toUpperCase());
+  // check if userenters an  empty input and clicks
+  if (isEmpty) {
+    alert('input is empty');
+  } else if (isDuplicate) {
+    alert('already exists');
+  } else {
+    if (EditTodoId >= 0) {
+      todos = todos.map((todo, index) => ({
+        ...todo,
+
+        value: index === EditTodoId ? todoValue : todo.value,
```

Comment:
The previous reviewer asked that you fix the edit functionality so that it does not add. a new todo item. Now you have a bug that i suggest you fix which does not allow me to edit the todos. the button does not work, nor does clicking the items themselves

https://user-images.githubusercontent.com/48546992/206829126-ede69595-b8ee-4b40-a071-dc5558ba3c50.mov

Reply:
@Nyame-Wolf its working...when you click on the edit button the todo pops at the input for editing. Kindly can you check that

Reply:
It still does not let me edit. Once i try to edit, and reload the application, the item still disappears meaning it's not saved in the local storage.